### ISSUE TITLE
Support for JMX

### ISSUE THREAD

**Issue by wayn3w** [voxpupuli/puppet-stash#79]
Would it be possible to support the JMX setting in the template et al?  There's both the stash JMX as well as the TomCat JMX we'd like to have enabled.

Three things would have to be added:
- CATALINA_OPTS would have to be exported in setenv.sh; this would define the necessary JMX environment variables
- 'jmx.enabled=' would have to be added to stash-config.properties
- an access file would have to be created specifying the 'monitorRole', the path of which would be specified in the aforementioned CATALINA_OPTS variable.

Is this something you'd be interested in supporting?

**Comment by mkrakowitzer** [comment]
Yeah I think this would be a useful feature and it should be added. Is this something you would consider submitting a pull request for?

**Comment by nibalizer** [comment]
+1 for this feature.

**Comment by wayn3w** [comment]
I can work on it and submit a pull request; it'll take me a little bit to get it done.

**Comment by mkrakowitzer** [comment]
Sounds good, let me know if you need any assistance. We have been waiting 2+ years for this feature so take as much time as you need :)

**Comment by claflico** [comment]
Rather than hard-coding variables & the template for each of the system properties (i.e. jdbc, jmx, setup automation) would it be better to make the entire template file able to be managed? You shouldn't have to define defaults for each of the variables as it sounds like they are already in stash and only need to be specified if you want to override the default stash value: https://confluence.atlassian.com/display/STASH/Stash+config+properties

Our devs are wanting to add this but I told them that custom properties weren't supported by the module yet and that I wasn't wanting to snowflake the server for it: https://confluence.atlassian.com/display/STASH/Stash+config+properties#Stashconfigproperties-Pullrequests

**Comment by mkrakowitzer** [comment]
The JMX configuration requires more than just changes to stash-config.properties files. The setenv.sh also need to be updated.

That being said we should allow a global parameter to all settings in  https://confluence.atlassian.com/display/STASH/Stash+config+properties to be configured. Can you raise a seperate issue for this? Its quick do to so ill work on it tonight.

**Comment by claflico** [comment]
done

**Comment by jyaworski** [comment]
Hello:

What is the status of this issue?
